src/mps/mpo.jl

```diff
@@ -1051,3 +1071,10 @@ function HDF5.read(parent::Union{HDF5.File,HDF5.Group}, name::AbstractString, ::
   v = [read(g, "MPO[$(i)]", ITensor) for i in 1:N]
   return MPO(v, llim, rlim)
 end
+
+function contract(::Algorithm"densitymatrix", M1::MPO, M2::MPO; kwargs...)::MPO
+  t1 = MPO([M1[v] for v in eachindex(M1)])
```

Comment:
I wanted to understand what this line is doing. Since `M1` is guarateed to be an MPO from the function signature, is `t1` needed?

Reply:
Thank you for pointing out this. This function is no more required as the new `contract` function accepts a pair of two AbstractMPS objects.